src/components/menu-item/menu-item.tsx

```diff
@@ -0,0 +1,468 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Fragment,
+  h,
+  Host,
+  Listen,
+  Method,
+  Prop,
+  State,
+  VNode
+} from "@stencil/core";
+import { FlipContext } from "../interfaces";
+import { Direction, getElementDir, slotChangeGetAssignedElements } from "../../utils/dom";
+import {
+  componentLoaded,
+  LoadableComponent,
+  setComponentLoaded,
+  setUpLoadableComponent
+} from "../../utils/loadable";
+import { CSS } from "./resources";
+import { MenuItemCustomEvent } from "./interfaces";
+
+type Layout = "horizontal" | "vertical";
+
+@Component({
+  tag: "calcite-menu-item",
+  styleUrl: "menu-item.scss",
+  shadow: true,
+  assetsDirs: ["assets"]
```

Comment:
should this component use delegatesFocus?

Reply:
It shouldn't . There isn't any non-focusable part in the component that users might interact with.

Reply:
nope, there isn't any non-focusable area in the menu-item.